com.unity.formats.fbx/Tests/FbxTests/ModelExporterTest.cs

```diff
@@ -731,26 +739,125 @@ public class SkinnedMeshTestDataClass
+                    var convertedPos = converted[k];
+
+                    var distance = getDistance(pos, convertedPos);
+                    if (distance < minDistance)
+                    {
+                        minDistance = distance;
+                    }
+                }
+                if (minDistance > maxDistance)
+                {
+                    maxDistance = minDistance;
+                }
+            }
+            return maxDistance;
+        }
+
+        private void TestFbxShapeNamesNotEmpty(FbxNode node)
+        {
+            var mesh = node.GetMesh();
+            if (mesh != null)
+            {
+                for (int i = 0; i < mesh.GetDeformerCount(); i++)
+                {
+                    var blendshape = mesh.GetBlendShapeDeformer(i);
+                    if (blendshape == null)
+                    {
+                        continue;
+                    }
+
+                    for(int j = 0; j < blendshape.GetBlendShapeChannelCount(); j++)
+                    {
+                        var blendShapeChannel = blendshape.GetBlendShapeChannel(j);
+                        for (int k = 0; k < blendShapeChannel.GetTargetShapeCount(); k++)
+                        {
+                            var shape = blendShapeChannel.GetTargetShape(k);
+                            Assert.That(string.IsNullOrEmpty(shape.GetName()), Is.False, string.Format("FbxShape missing name on blendshape {0}", blendshape.GetName()));
+                        }
+                    }
+                }
+            }
+
+            for (int i = 0; i < node.GetChildCount(); i++)
+            {
+                TestFbxShapeNamesNotEmpty(node.GetChild(i));
+            }
+        }
+
+        private void FbxImportAndTestBlendshapes(string fbxPath)
         {
-            public int Compare(Vector3 a, Vector3 b)
+            // Create the FBX manager
+            using (var fbxManager = FbxManager.Create())
             {
-                Assert.That(a.x, Is.EqualTo(b.x).Within(0.00001f));
-                Assert.That(a.y, Is.EqualTo(b.y).Within(0.00001f));
-                Assert.That(a.z, Is.EqualTo(b.z).Within(0.00001f));
-                return 0;  // we're almost equal
+                FbxIOSettings fbxIOSettings = FbxIOSettings.Create(fbxManager, Globals.IOSROOT);
+
+                // Configure the IO settings.
+                fbxManager.SetIOSettings(fbxIOSettings);
+
+                // Create the importer 
+                var fbxImporter = FbxImporter.Create(fbxManager, "Importer");
+
+                // Initialize the importer.
+                int fileFormat = -1;
+
+                bool status = fbxImporter.Initialize(fbxPath, fileFormat, fbxIOSettings);
+                FbxStatus fbxStatus = fbxImporter.GetStatus();
+
+                Assert.That(status, Is.True, fbxStatus.GetErrorString());
+                Assert.That(fbxImporter.IsFBX(), "file does not contain FBX data");
```

Comment:
is this an `Assert.True` with an error message if the condition is false?

Reply:
Yes that's right